autopush/websocket.py

```diff
@@ -365,15 +368,31 @@ def _check_other_nodes(self, result):
             attrs = result.get("Attributes")
             node_id = self._get_aval(attrs.get("node_id", {}), 'S')
             uaid = self._get_aval(attrs.get("uaid", {}), 'S')
-            connected_at = self._get_aval(attrs.get("connected_at", {}), 'N')
-            url = "%s/notif/%s/%s" % (node_id, uaid, connected_at)
-            if not getattr(self, 'testing', False):
-                d = self.ap_settings.agent.request(
-                    "DELETE",
-                    url.encode("utf8"),
-                ).addCallback(self.finish_hello)
-                d.addErrback(self.log_err, extra="Failed to delete old node")
-                return
+            last_connect = self._get_aval(attrs.get("connected_at", {}), 'N')
+            existing = self.ap_settings.clients.get(uaid)
+            if existing:
+                if existing.connected_at <= last_connect:
+                    existing.sendClose()
+                    return
+                if existing.connected_at > last_connect:
```

Comment:
The `last_connect` is the last, recorded connection we have. 
`existing` is the known existing connection. 
There's also the current connection, which isn't mentioned here, but was just written to the database.

If the existing connection is the same age or older than the last recorded connection, we close it in favor of the current connection; the current connection "wins"
If the existing connection is newer than the last recorded connection, we close the existing connection and return, preventing the current connection from completing as well; the last recorded connection "wins".

Currently, there is no case where the "existing" connection wins, however there's additional logic inside of router to potentially reject connections.

I agree that this is becoming increasingly complicated and that there is a division of logic.